shield.go

```diff
@@ -17,6 +18,8 @@ const MaxLength = 4096
 // SaltLength is the length of the salt generated by GenerateSalt.
 const SaltLength = 32
 
+var ErrPasswordTooLong = errors.New("password too long")
```

Comment:
Another possibility would be to declare error like this:

```go
type Err string

func (e Err) Error() string { return string(e) }
```

And then create the constant:

```go
const err = Err("lorem ipsum")
```

This works as expected, however, note that in the stdlib the default way is declaring it as variables instead: [`database/sql.ErrNoRows`](https://golang.org/src/database/sql/sql.go#L210) or [`io.EOF`](https://golang.org/src/io/io.go#L38). It makes sense, as `errors.New` returns a pointer.